Added helper variable, removed extra slash

juzraai opened the pull request:
`category_root` variable helps layout to decide whether to render a category list or a post list - without knowing the title of the category list page (old version). This way, title of the category list can be
edited at only one place. :)

Also there was an extra slash in taxonomy link which produced a "/categories//category-name/" like URL.

Awesome plugin anyway! :)

juzraai committed:
Added helper variable, removed extra slash

`category_root` variable helps layout to decide whether to render a
category list or a post list - without knowing the title of the category
list page (old version). This way, title of the category list can be
edited at only one place. :)

juzraai committed:
Restoring accidentally deleted lines

Lewiscowles1986 commented:
This looks promising, I take it you assign `category_root`; or is it a jekyll feature?

The other question I have is which versions of Jekyll this has been tested up to and what config.yml looks like in the case you have two slashes?

Lewiscowles1986 commented on _layouts/category_index.html:
```diff
@@ -2,13 +2,10 @@
 	{% include taxonomy_sort.html source=site.categories %}
 	{% for taxonomy in sortedtaxonomies %}
-		<h2><a href="{{ page.url | prepend: base_url | replace: "/index.html", "" }}/{{ taxonomy }}/">{{ taxonomy }}</a></h2>
+		<h2><a href="{{ page.url | prepend: base_url | replace: "/index.html", "" }}{{ taxonomy }}/">{{ taxonomy }}</a></h2>
```
This is the line that slightly concerns me. As we are replacing `/index.html` with `""`, we should (at least in my tiny head) need to prefix a `/`

This hints at the `/` part being a config, so I'd like to check on that before merging

juzraai committed:
Added slash appending and double slash replacing

Adding a slash is needed in Jekyll v2.3.5 but in Jekyll v3.1.6 it causes
a double slash in the URL. This append+replace fixes this.

juzraai commented:
> This looks promising, I take it you assign category_root; or is it a jekyll feature?

I assigned it :) in `category_pages.rb` on lines 19 and 22.


> which versions of Jekyll this has been tested up to and what config.yml looks like in the case you have two slashes?

I used the default config.yml:

```
title: Your awesome title
email: your-email@domain.com
description: ...
baseurl: ""
url: "http://yourdomain.com"
markdown: kramdown 
```

With Jekyll v3.1.6 and your code, the generated HTML looks like this:

```html
<ul class="posts">
	<h2><a href="/categories//jekyll/">jekyll</a></h2>
	<h2><a href="/categories//update/">update</a></h2>
</ul>	
```

But now I've tried with Jekyll v2.3.5, and here your code works well:

```html
<ul class="posts">
	<h2><a href="/categories/jekyll/">jekyll</a></h2>
	<h2><a href="/categories/update/">update</a></h2>
</ul>
```

Strange... my code fixes double slash in Jekyll v3.1.6 but creates a mess in v2.3.5.

I've just added a new commit which uses Liquid filters to add the slash, and then replace double slash with a single one. I tested it in both Jekyll v3.1.6 and v2.3.5, it works well:

```
<h2><a href="{{ page.url | prepend: base_url | replace: "/index.html", "" | append: "/" | replace: "//", "/" }}{{ taxonomy }}/">{{ taxonomy }}</a></h2>
```

Lewiscowles1986 commented:
Thank you very much :+1: . It's true, I think I wrote this pre 3.x :wink:
